packages/reactive-element/src/css-tag.ts

```diff
@@ -189,6 +185,58 @@ export const adoptStyles = (
   }
 };
 
+/**
+ * Applies the given styles to a `shadowRoot` when `adoptedStyleSheets` is available and returns `true`.
+ * Otherwise, it does nothing and returns `false`.
+ */
+export const adoptStyleSheets = (
+  renderRoot: ShadowRoot,
+  styles: Array<CSSResultOrNative>
+) => {
+  if (supportsAdoptingStyleSheets) {
+    const sheets = styles.map(
+      (result) => (result as CSSResult).styleSheet ?? result
```

Comment:
To reduce size, let's try inlining the fix right here if and only if the sheet instanceof root.ownerDocument.defaultView.CSSStyleSheet is false. Let's also try not doing the caching.

Reply:
Sure...Will try that.

Reply:
In light of @justinfagnani's suggestion, should I scrap this, or should I go on?